parserator_web/views.py

```diff
@@ -16,9 +16,33 @@ class AddressParse(APIView):
     def get(self, request):
         # TODO: Flesh out this method to parse an address string using the
         # parse() method and return the parsed components to the frontend.
-        return Response({})
+        address = request.query_params.get("address")
+        input_string = str(address)
+        try:
+            address_components, address_type = self.parse(input_string)
+            return Response({
+                'Address': input_string,
+                'Address Type': address_type,
+                'Address Components': address_components,
+                }, status=200)
+        except:
+            original_string, parsed_string, error_desc = self.parse(input_string)
+            
+            return Response({
+                'Original String': original_string,
+                'Parsed String': parsed_string,
+                'Error Description': error_desc,
+            }, status=400)
+
 
     def parse(self, address):
         # TODO: Implement this method to return the parsed components of a
         # given address using usaddress: https://github.com/datamade/usaddress
+        try:
+            address_components, address_type = usaddress.tag(address)[0], usaddress.tag(address)[1]
+
+        except usaddress.RepeatedLabelError as e:
+            error_desc = "Not a valid address!"
+            return e.original_string, e.parsed_string, error_desc
```

Comment:
I think it's a little confusing to return different things from a method, depending on whether the method succeeds. Thoughts on raising this error and letting the `get()` method decide what to do in your try/except block, instead of returning the error details here?

Reply:
Sure, I think that makes a lot of sense! But I had trouble figuring out how to raise the error while still passing along information to the front end, so it still returns something for now in the `get()` method instead of using `raise`. I would love to learn better ways to handle this! a6f3271c0ac4189972819d9eda4570ba921fbd9e